### DISCUSSION TITLE
results.json not generated

### DISCUSSION THREAD

**Discussion by erlebach** [gradescope/autograder_samples#115] [Q&A] [answered]
I have created my auto-grading system for a programming assignment with testing via pytest. My tests run correctly when using run_autograder/ on the Gradescope server. However, the file `results.json` is not generated. I am pretty confident that this file is generated through a call to `harness.py`, but I cannot figure out who/what initiates the call. What must I do to guarantee the creation of a `results.json` file? If I do everything correctly, of course, the file is created. There is remarkably sparse information on the web regarding what to do if it is not created. The creation process is not explained anywhere that I have found as yet. Thanks for any help.

**Comment by adam-brucker** [comment, marked as answer]
If you have not watched the autograder video that would be a good starting point.  https://youtu.be/ZX3G5dFRZKI?si=92kS5QK8bfTOs7zi&t=2384
